bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/ThingConsoleCommandExtension.java

```diff
@@ -140,6 +154,94 @@ private void printThings(Console console, Collection<Thing> things) {
         }
     }
 
+    private void printThingsDetails(Console console, List<String> thingUIDStrings) {
+        Collection<Thing> things;
+
+        if (thingUIDStrings.isEmpty()) {
+            things = thingRegistry.getAll();
+        } else {
+            things = new ArrayList<>();
+            for (String thingUIDString : thingUIDStrings) {
+                ThingUID thingUID;
+                try {
+                    thingUID = new ThingUID(thingUIDString);
+                } catch (IllegalArgumentException e) {
+                    console.println("This is not a valid thing UID: " + thingUIDString);
+                    return;
+                }
+
+                Thing thing = thingRegistry.get(thingUID);
+                if (thing == null) {
+                    console.println("Could not find thing with UID " + thingUID);
+                    return;
+                }
+                things.add(thing);
+            }
+        }
+
+        printThingsDetails(console, things);
+    }
+
+    private void printThingsDetails(Console console, Collection<Thing> things) {
+        for (Iterator<Thing> iter = things.iterator(); iter.hasNext();) {
+            printThingDetails(console, iter.next());
+            if (iter.hasNext()) {
+                console.println("");
+                console.println("--- --- --- --- ---");
+                console.println("");
+            }
+        }
+    }
+
+    private void printThingDetails(Console console, Thing thing) {
+        console.println("UID: " + thing.getUID());
+        console.println("Type: " + thing.getThingTypeUID());
+        console.println("Label: " + thing.getLabel());
+        console.println("Status: " + thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(thing, null));
+        if (thing.getBridgeUID() != null) {
+            console.println("Bridge: " + thing.getBridgeUID());
+        }
+        console.println("");
+
+        if (thing.getProperties().isEmpty()) {
+            console.println("No properties");
+        } else {
+            console.println("Properties:");
+            for (Map.Entry<String, String> entry : thing.getProperties().entrySet()) {
+                console.println("\t" + entry.getKey() + " : " + entry.getValue());
+            }
+        }
+        console.println("");
+
+        if (thing.getConfiguration().getProperties().isEmpty()) {
+            console.println("No configuration properties");
```

Comment:
wrt handler configuration we usually use the term `parameter` instead of property. See [here](https://www.eclipse.org/smarthome/documentation/development/bindings/thing-handler.html#handling-configuration-updates) for reference.

Reply:
Thanks for the hint! Learned something new, always nice :)